frame/evm/src/lib.rs

```diff
@@ -761,7 +762,15 @@ impl<T: Config> Pallet<T> {
 		let digest = <frame_system::Pallet<T>>::digest();
 		let pre_runtime_digests = digest.logs.iter().filter_map(|d| d.as_pre_runtime());
 
-		T::FindAuthor::find_author(pre_runtime_digests).unwrap_or_default()
+		if let Some(author) = T::FindAuthor::find_author(pre_runtime_digests) {
```

Comment:
I think you should implemente your own `FindAuthor` rather than hard code it in the lib.
How about adding a `from_account_id` method for `AddressMapping`, so we can implement the custom `FindAuthor` like this:

```rust
pub struct MyFindAuthor<F, M>(PhantomData<(F, M)>);
impl<F: FindAuthor<u32>, M: AddressMapping<AccountId>> FindAuthor<H160> for MyFindAuthor<F, M> {
	fn find_author<'a, I>(digests: I) -> Option<H160>
	where
		I: 'a + IntoIterator<Item = (ConsensusEngineId, &'a [u8])>,
	{
		if let Some(author_index) = F::find_author(digests) {
			let authority_id = Aura::authorities()[author_index as usize].clone();
			let eth_address: H160 = M::from_account_id(AccountId::from_slice(&authority_id.to_raw_vec()).unwrap());
            Some(eth_address)
		}
		None
	}
}
```
in the runtime, just add:

```rust
type FindAuthor = MyFindAuthor<Aura, PairedAddressMapping<Self>>;
```

In this way we don't need to change too much `pellet-evm` source code, just modified `AddressMapping`, we can convert substrate address and evm address not only in `pallet-evm`, but also in runtime, or future other pallet.